### PR TITLE
(PC-5854) Add test to verify that all analytics event respect firebas…

### PR DESCRIPTION
**Pull request by BenStirrup**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-5854

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

### CODE REVIEW

File: src/libs/analytics.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+export {}
+
+jest.mock('@react-native-firebase/analytics')
+
+const { ANALYTICS_EVENT } = jest.requireActual('./analytics')
+
+describe('analytics', () => {
+  describe('AnalyticsEvent', () => {
+    it('actual members should respect Firebase naming rules', () => {
```
Comment (antoinewg) [inline src/libs/analytics.test.ts:9]:
```js
    it('actual members should respect Firebase naming rules', () => {
      const eventNames = Object.values(ANALYTICS_EVENT) as string[]
      eventNames.forEach((eventName) => {
        const isValidMember = validateAnalyticsEvent(eventName)
        expect(isValidMember).toBeTruthy()
      })
    })
```

si tu garde l'enum, ça marche bien non?

File: src/libs/analytics.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      for (const member of Object.values(ANALYTICS_EVENT)) {
+        const isValidMember = validateAnalyticsEvent(member as string)
+        if (!isValidMember) {
+          fail(`${member} is not valid`)
+        }
+      }
+    })
+    it('fake member should fail the test when it is too long', () => {
+      const isValidMember = validateAnalyticsEvent('12345678901234567890123456789012345678901')
```
Comment (antoinewg) [inline src/libs/analytics.test.ts:18]:
```js
    it('fake member should fail the test when it is too long', () => {
      expect(validateAnalyticsEvent('a'.repeat(39))).toBeTruthy()
      expect(validateAnalyticsEvent('a'.repeat(40))).toBeTruthy()
      expect(validateAnalyticsEvent('a'.repeat(41))).toBeFalsy()
    })
```

File: src/libs/analytics.test.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    })
+    it('fake member should fail the test when it contains a non alpha numeric char', () => {
+      const isValidMember = validateAnalyticsEvent('Event-Testé')
+      expect(isValidMember).toBeFalsy()
+    })
+    it('fake member should fail the test when it includes a "-"', () => {
+      const isValidMember = validateAnalyticsEvent('Event-Test')
+      expect(isValidMember).toBeFalsy()
+    })
+    it('fake member should fail the test when it includes a reserved keyword', () => {
```
Comment (antoinewg) [inline src/libs/analytics.test.ts:37]: je trouve que c'est plus lisible en une ligne:

```js
    it('fake member should fail the test when it includes a reserved keyword', () => {
      expect(validateAnalyticsEvent('firebase_test')).toBeFalsy()
      expect(validateAnalyticsEvent('google_test')).toBeFalsy()
      expect(validateAnalyticsEvent('ga_test')).toBeFalsy()
      expect(validateAnalyticsEvent('test_firebase_test')).toBeTruthy()
    })
```

File: src/libs/analytics.test.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      expect(validateAnalyticsEvent('firebase_test')).toBeFalsy()
+      expect(validateAnalyticsEvent('google_test')).toBeFalsy()
+      expect(validateAnalyticsEvent('ga_test')).toBeFalsy()
+    })
+  })
+})
+
+const RESERVED_KEYWORDS = ['firebase_', 'google_', 'ga_']
```
Comment (antoinewg) [inline src/libs/analytics.test.ts:41]:
```suggestion
const RESERVED_PREFIXES = ['firebase_', 'google_', 'ga_']
```

c'est pour ça que j'ai rajouté `expect(validateAnalyticsEvent('test_firebase_test')).toBeTruthy()` comme test

**Review by antoinewg — APPROVED** [review]

**Comment by BenStirrup** [comment]
A merger après la fin du freeze merge

**Pull request merged**